src/plugin/ethTransaction.ts

```diff
@@ -717,8 +779,8 @@ export class EthereumTransaction implements Interfaces.Transaction {
     const signedTrx = ethJsTx.sign(privateKeyBuffer)
     const signature = {
       v: bufferToInt(signedTrx.v?.toBuffer()),
-      r: signedTrx.r?.toBuffer(),
-      s: signedTrx.s?.toBuffer(),
+      r: bufferPadStart(signedTrx.r?.toBuffer(), 32), // The validate signature logic expects 32 bytes. Some r and s values can result in 31 bytes - pad the number to ensure it's seen as valid.
+      s: bufferPadStart(signedTrx.s?.toBuffer(), 32), // The validate signature logic expects 32 bytes. Some r and s values can result in 31 bytes - pad the number to ensure it's seen as valid.
     } as EthereumSignatureNative
```

Comment:
we should not be forcing the type here (as EthereumSignatureNative) - use toEthereumSignatureNative() thats why we have it.
Remove the bufferPadStart code here - it is in toEthereumSignatureNative

Reply:
Same comment as above. The focus of this PR is not to resolve bad signature code. This patch was put in place as it was blocking my work on this PR.